core/chains/starknet/chain.go

```diff
@@ -2,57 +2,121 @@ package starknet
 
 import (
 	"context"
+	"math"
+	"math/rand"
+	"time"
 
+	caigotypes "github.com/dontpanicdao/caigo/types"
+	"github.com/pkg/errors"
+
+	starkChain "github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/chain"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/config"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/db"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/txm"
 	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/starknet"
-	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/starknet/db"
 
 	"github.com/smartcontractkit/sqlx"
 
 	"github.com/smartcontractkit/chainlink/core/chains/starknet/types"
 	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
-var _ starknet.Chain = (*chain)(nil)
+var _ starkChain.Chain = (*chain)(nil)
 
 type chain struct {
 	utils.StartStopOnce
 	id   string
-	cfg  starknet.Config
+	cfg  config.Config
 	orm  types.ORM
 	lggr logger.Logger
+	txm  txm.StarkTXM
 }
 
-func NewChain(db *sqlx.DB, dbchain types.DBChain, orm types.ORM, lggr logger.Logger) (*chain, error) {
-	cfg := starknet.NewConfig(*dbchain.Cfg, lggr)
+func NewChain(db *sqlx.DB, ks keystore.StarkNet, dbchain types.DBChain, orm types.ORM, lggr logger.Logger) (ch *chain, err error) {
+	cfg := config.NewConfig(*dbchain.Cfg, lggr)
 	lggr = lggr.With("starknetChainID", dbchain.ID)
-	var ch = chain{
+
+	ch = &chain{
 		id:   dbchain.ID,
 		cfg:  cfg,
 		orm:  orm,
 		lggr: lggr.Named("Chain"),
 	}
 
-	return &ch, nil
+	getClient := func() (caigotypes.Provider, error) {
+		return ch.getClient()
+	}
+
+	ch.txm, err = txm.New(lggr, ks, cfg, getClient)
+	if err != nil {
+		return nil, err
+	}
+
+	return ch, nil
 }
 
-func (c *chain) Config() starknet.Config {
+func (c *chain) Config() config.Config {
 	return c.cfg
 }
 
 func (c *chain) UpdateConfig(cfg *db.ChainCfg) {
 	c.cfg.Update(*cfg)
 }
 
+func (c *chain) TxManager() txm.TxManager {
+	return nil
+}
+
+func (c *chain) Reader() (starknet.Reader, error) {
+	return nil, nil
```

Comment:
Also missing implementation, is this used? We should think more on Client construction as mentioned by Blaz in another comment.

Reply:
yeah, my bad I forgot to connect it the `c.getClient`